include/chainbase/pinnable_mapped_file.hpp

```diff
@@ -23,6 +23,8 @@ class pinnable_mapped_file {
       };
 
       pinnable_mapped_file(const bfs::path& dir, bool writable, uint64_t shared_file_size, bool allow_dirty, map_mode mode, std::vector<std::string> hugepage_paths);
+      pinnable_mapped_file(pinnable_mapped_file&&) = default;
+      pinnable_mapped_file& operator=(pinnable_mapped_file&&) = default;
```

Comment:
I don't see how the default move constructor and move assignment would do the right thing here with the `_writable`, `_segment_manager`, etc. Seems like the moved from object's destructor would do "bad things" to the moved to object.

Reply:
What's the issue with `_writeable` that's just a bool, default move operator should be fine right?

`_segment_manager` is tricky. pinnable_mapped_file does not own it. It is owned either by the `_mapped_file` (in the case the user is operating in mapped mode) or `_mapped_region` (in the case user is operating in heap or lockec mode). For example pinnable_mapped_file will never delete it because destruction of `_mapped_region`/`_mapped_file` will release that resource that is internal to them. In some ways this is an optimization of sorts: I didn't want a branch in `get_segment_manager()` that has to determine if it should return the segment manager of the `_mapped_region` or `_mapped_file` based on whichever one is in use.

Reply:
My point about `_writeable` is that it would remain true. But I see now that since `_mapped_region` would be in a default constructed state, that you should be fine.

Reply:
ah, no, I think you're right there might be some problems in the dtor there. I've got another PR coming today on chainbase I'll clean up the dtor there I think